bindings/java/mongocrypt/src/main/java/com/mongodb/crypt/capi/MongoCryptImpl.java

```diff
@@ -96,37 +98,51 @@
             }
         }
 
+        if (options.getLocalSchemaMap() != null) {
+            BsonDocument localSchemaMapDocument = new BsonDocument();
+            for (Map.Entry<String, BsonDocument> cur: options.getLocalSchemaMap().entrySet()) {
+                localSchemaMapDocument.put(cur.getKey(), cur.getValue());
+            }
+
+            mongocrypt_binary_t localSchemaMapBinary = toBinary(localSchemaMapDocument);
+            try {
+                success = mongocrypt_setopt_schema_map(wrapped, localSchemaMapBinary);
+                if (!success) {
+                    throwExceptionFromStatus();
+                }
+            } finally {
+                mongocrypt_binary_destroy(localSchemaMapBinary);
+            }
+        }
+        
         success = mongocrypt_init(wrapped);
         if (!success) {
             throwExceptionFromStatus();
         }
     }
 
     @Override
-    public MongoCryptContext createEncryptionContext(final String namespace, final BsonDocument localSchemaDocument) {
+    public MongoCryptContext createEncryptionContext(final String database, final BsonDocument commandDocument) {
         isTrue("open", !closed);
+        notNull("database", database);
+        notNull("commandDocument", commandDocument);
         mongocrypt_ctx_t context = mongocrypt_ctx_new(wrapped);
         if (context == null) {
             MongoCryptContextImpl.throwExceptionFromStatus(context);
```

Comment:
Unrelated, but if the new context is NULL, that's an error on the mongocrypt_t handle. I think this should just be:

```
throwExceptionFromStatus();
```

Reply:
I'll fix it.